src/Components/Artwork/SaveButton/SaveArtworkToListsButton.tsx

```diff
@@ -15,19 +20,58 @@ const SaveArtworkToListsButton: FC<SaveArtworkToListsButtonProps> = ({
   artwork,
   contextModule,
 }) => {
-  const { isSavedToList, savedListId } = useManageArtworkForSavesContext()
-  const { isSaved, onSave } = useSaveArtworkToLists({
-    artwork,
+  const tracking = useTracking()
+  const { savedListId, isSavedToList } = useManageArtworkForSavesContext()
+
+  const customListsCount = artwork.customCollections?.totalCount ?? 0
+  const isSavedToDefaultList = !!artwork.is_saved
+  const isSavedToCustomLists = customListsCount > 0
+
+  const { isSaved, saveArtworkToLists } = useArtworkLists({
     contextModule,
-    customListsCount: artwork.customCollections?.totalCount ?? 0,
-    isSaved: !!artwork.is_saved,
+    artwork: {
+      internalID: artwork.internalID,
+      id: artwork.id,
+      slug: artwork.slug,
+      title: `${artwork.title}, ${artwork.date}`,
+      imageURL: artwork.preview?.url ?? null,
+      isSavedToDefaultList,
+      isSavedToCustomLists,
+    },
   })
 
-  const handleClick = (
+  const getActionLabel = (action: ResultAction) => {
```

Comment:
**suggestion (non-blocking):** A couple of readability thoughts…

Type this arg more narrowly, e.g. 

```ts
action: ResultAction.SavedToDefaultList | ResultAction.RemovedFromDefaultList
```

Make the mapping clearer:

```ts
  return {
    [ResultAction.SavedToDefaultList]: "Saved Artwork",
    [ResultAction.RemovedFromDefaultList]: "Removed Artwork",
  }[action]
```

Reply:
Done: https://github.com/artsy/force/pull/12116/commits/1e0ef77e8b017908c138718e73f6110b2190c126